e2e-tests/src/test/button_game/mod.rs

```diff
@@ -487,28 +487,30 @@ async fn button_game_play<F: Fn(u128, u128, u128, u128)>(
     sleep(Duration::from_secs(3)).await;
 
     button.press(&sign(&conn, player)).await?;
-    let event = assert_recv_id(&mut events, "ButtonPressed").await;
-    let_assert!(Some(&Value::UInt(second_presser_score)) = event.data.get("score"));
+    let event = assert_recv_id(&mut events, "RewardMinted").await;
+    let_assert!(Some(&Value::UInt(second_reward_amount)) = event.data.get("amount"));
     let_assert!(Some(&Value::UInt(second_press_at)) = event.data.get("when"));
     let first_presser_time = first_press_at - reset_at;
     let second_presser_time = second_press_at - first_press_at;
 
     score_check(
         first_presser_time,
-        first_presser_score,
+        first_reward_amount,
         second_presser_time,
-        second_presser_score,
+        second_reward_amount,
     );
-    let total_score = first_presser_score + second_presser_score;
-    assert!(reward_token.balance_of(&conn, player.account_id()).await? == total_score);
+
+    let player_rewards = first_reward_amount + second_reward_amount;
+    assert!(reward_token.balance_of(&conn, player.account_id()).await? == player_rewards);
 
     wait_for_death(&conn, &button).await?;
     button.reset(&sign(&conn, &authority)).await?;
-    assert_recv_id(&mut events, "Reset").await;
+    let event = assert_recv_id(&mut events, "RewardMinted").await;
+    let_assert!(Some(&Value::UInt(pressiah_reward)) = event.data.get("amount"));
 
-    let pressiah_score = total_score / 4;
```

Comment:
This is a somewhat different test than before - now we depend on the events to tell us what's the reward of the pressiah, before we asserted that it's some specific number. Are we OK with that?

Reply:
It's just that the mapping of scores to the domain of rewards does not preserve that fact. 
(The score is a quarter of the totals but the reward (after mapping to a fixed domain) is not). 

tl;dr: given it would be a big overhaul to test that, and all we get is a test of dividing by 4, imo not worth the effort